source/exe/main_common_base.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include "envoy/event/timer.h"
+#include "envoy/runtime/runtime.h"
+#include "envoy/server/platform.h"
+
+#include "source/common/common/thread.h"
+#include "source/common/event/real_time_system.h"
+#include "source/common/grpc/google_grpc_context.h"
+#include "source/common/stats/symbol_table.h"
+#include "source/common/stats/thread_local_store.h"
+#include "source/common/thread_local/thread_local_impl.h"
+#include "source/exe/process_wide.h"
+#include "source/server/listener_hooks.h"
+#include "source/server/options_impl.h"
+#include "source/server/server.h"
+
+#ifdef ENVOY_HANDLE_SIGNALS
+#include "source/common/signal/signal_action.h"
+#include "source/exe/terminate_handler.h"
+#endif
+
+namespace Envoy {
+
+class ProdComponentFactory : public Server::ComponentFactory {
+public:
+  // Server::DrainManagerFactory
+  Server::DrainManagerPtr createDrainManager(Server::Instance& server) override;
+  Runtime::LoaderPtr createRuntime(Server::Instance& server,
+                                   Server::Configuration::Initial& config) override;
+};
+
+class MainCommonBase {
```

Comment:
Can this class be documented? e.g. especially differences between MainCommon and MainCommonBase